website/docs/developer/development-guide/code/integration-tests.md

```diff
@@ -0,0 +1,163 @@
+- All resources, including storage mediums, should be deleted after all tests have been run.
+  :::warning
+  KMS keys persist for a minimum of 7 days. This can be gotten around by randomizing the environment's prefix.
+  :::
+- For initial setup, we do not plan on automating checking logs.
+  In the future, this may be a method of identifying point-of-failure in automated processes.
+- Files should be run on a regular cadence. Initial suggestion is once every 1-2 weeks.
+
+Some broad categories of tests are shown below.
+
+### Happy Path
+
+These are basic tests that perform a full run-through of a feature.
+Randomized data is used to create a valid state, and a valid user is used to trigger/retrieve work.
+
+### Security Paths
+
+If feasible, there should be a copy of each [Happy Path](#happy-path) test that checks the same route, but with one property modified so as to fail security checks.
+For lambdas with properly specified permissions, this will involve calling the lambda with a user that does not have permission to invoke the lambda.
+For components such as databases and S3 buckets, this will involve attempting to retrieve/delete/add data with a user that does not have the appropriate permissions.
+
+:::tip
+To improve maintainability, use a shared function for setup of the [Happy Path](#happy-path) and Security Path tests for a given component.
+:::
+
+:::warning
+Due to how NGAP handles security, even resources with public access enabled cannot be accessed publicly.
+Further research should be done to identify how to perform these tests.
+:::
+
+### Error Paths
+
+These tests verify that when an error is expected, a proper error is returned/raised.
+For example, several entries in our [API](../../../developer/api/api-gateway.md) should properly return a 404 error code if the entry does not exist in the database.
+The test should fail if the API returns the error in a dictionary instead of an HTTP status code, or the API returns any other error code.
+
+## What Not to Test
+
+### External Integrations
+
+We do not presently test integrations with Cumulus or other external consumers.
+As we do not want to deepen coupling with Cumulus, it is best to focus on maintaining a consistent API.
+
+### Performance
+
+While there is a desire to eventually develop performance tests, these tests should focus on functionality with generous timeouts.
+
+## Needed Tests
+
+This is a list of tests that should be created for existing Orca architecture. This list may change as tests are created and components are modified.
+
+### [Internal Reconciliation](../../research/research-reconciliation.mdx)
+- [Happy](#happy-path):
+  1. Ingest randomized data to Orca.
+  1. Modify the catalog and post S3 data in the structure of an S3 inventory report to the report bucket. Include at least one of each error type comparing between the two sources.
```

Comment:
Put a note that we want a pre-created S3 data report because we do not want to wait for 24 hours for one to be created. We should also put some notes of running the S3 report through AWS mechanisms to validate structure and content periodically to validate our reporting test.

Reply:
Former: We have not created the dummy report yet. That will be part of implementing the test.
Later: Changed.